consensus/src/sync/live/block_queue/queue.rs

```diff
@@ -216,17 +207,32 @@ impl<N: Network> BlockQueue<N> {
                 );
                 return Some(QueuedBlock::TooFarAhead(peer_id));
             }
-        } else if block_number <= macro_height {
+        } else if block_number <= self.current_macro_height {
```

Comment:
It's safer to use `macro_height` here (computed as before: `Policy::last_macro_block(head_height)`), as otherwise there is a race condition between `check_and_prune()` (which updates `self.current_macro_height`) and `check_announced_block()`, because the blockchain read lock is dropped between those calls.